mixnet/protocol/temporalmix.py

```diff
@@ -0,0 +1,145 @@
+import random
+from abc import abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import TypeVar
+
+from framework.framework import Framework, Queue
+
+
+class TemporalMixType(Enum):
+    NONE = "none"
+    PURE_COIN_FLIPPING = "pure-coin-flipping"
+    PURE_RANDOM_SAMPLING = "pure-random-sampling"
+    PERMUTED_COIN_FLIPPING = "permuted-coin-flipping"
+
+
+@dataclass
+class TemporalMixConfig:
+    mix_type: TemporalMixType
+    # The minimum size of queue to be mixed.
+    # If the queue size is less than this value, noise messages are added.
+    min_queue_size: int
+    # Generate the seeds used to create the RNG for each queue that will be created.
+    seed_generator: random.Random
```

Comment:
For reproducibility purposes this should be fixed to some value. Otherwise we are going to have different results every run.

Reply:
And getting different results for each iteration/run is what we want. So maybe we should use the same seed for every session of experiments and simply add the iteration number to it. Then we are going to have reproducibility for each session and different results for each iteration/run.

Reply:
Yes, exactly. I use the iteration number as a seed, because we expect to have different results between iterations, as you mentioned. You can find this in the [next PR](https://github.com/logos-co/nomos-simulations/pull/12/files#diff-7b496529b8597edfe762c7ad7bcde43e0a44c608d4e9f4b04628ddecf718c6b6R134). The results of all iterations are aggregated to calculate min, avg, max, std, etc...
So, if we re-run the session 1 experiment, we will get the 100% same result.